feat: add clonebuilder and test cases #8 #9

jonasjungaker opened the pull request:
Added functionality for building a project at a path with a maven command line interface. the checkBuildOutput method returns a builddata instance of the information we need.

jonasjungaker committed:
feat: add clonebuilder and test cases #8 #9

Added functionality for building a project at a path with a maven command line interface. the checkBuildOutput method returns a builddata instance of the information we need.

jonasjungaker committed:
Merge branch 'master' into feat/cloneBuilder

jvonhacht commented:
Add docstrings :)

jvonhacht reviewed (changes requested):
Add docstrings

jonasjungaker committed:
doc: add docstrings #27

Added docstrings to files for compiling using maven and the BuildData object class.

jonasjungaker committed:
Merge branch 'feat/cloneBuilder' of https://github.com/jonasjungaker/DD2480_A2_CI_Group10 into feat/cloneBuilder

jvonhacht commented on ci/src/main/java/group10/CloneBuilder.java:
```diff
@@ -0,0 +1,52 @@
+            int beginIndex = buildString.indexOf("[ERROR] Failures:");
+            int endIndex = buildString.indexOf("[INFO] BUILD FAILURE");
+            String message = "Test failure at build-time with errors:\n".concat(buildString.substring(beginIndex, endIndex));
+            return new BuildData(true, false, message);
```
If the input buildString is empty for some reason, there will be an indexOutOfBounds exception

jvonhacht commented on ci/src/main/java/group10/BuildData.java:
```diff
@@ -0,0 +1,18 @@
+ * A class for containing information about builds with maven
+ * For handling specific build ouputs
+ */
+public class BuildData
```
Consider placing this in the util folder or is that confusing?

jonasjungaker committed:
refactor: create a more complete Clonebuilder #27

Added a more complete clonebuilder and removed the builddata class for conciseness and readability
